pyproject.toml

```diff
@@ -1,19 +1,117 @@
-[tool.black]
-exclude = "3rdparty/*"
-line-length = 100
+[project]
+name = "xgrammar"
+description = "Efficient, Flexible and Portable Structured Generation"
+authors = [{ name = "MLC Team" }]
+readme = "README.md"
+license = { text = "Apache 2.0" }
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Education",
+    "Intended Audience :: Science/Research",
+]
+keywords = ["machine learning", "inference"]
+requires-python = ">=3.9, <4"
+dependencies = [
+    "pydantic",
+    "sentencepiece",
+    "tiktoken",
+    "torch>=2.5.0",
+    "transformers>=4.38.0",
+]
+dynamic = ["version"]
 
-[tool.isort]
-profile = "black"
-src_paths = ["python"]
+[project.urls]
+Homepage = "https://xgrammar.mlc.ai/"
+GitHub = "https://github.com/mlc-ai/xgrammar"
 
-[tool.mypy]
-strict = true
+[project.optional-dependencies]
+triton = [
+    "triton",
```

Comment:
Just add triton as a normal dependency
```
"triton; platform_system == 'Linux' and platform_machine == 'x86_64'"
```

Reply:
Sure, but it looks like the code is favoring the direct Cuda version and not the trition version now since you have to set an env to even use triton.